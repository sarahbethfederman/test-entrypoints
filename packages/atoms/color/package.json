{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  