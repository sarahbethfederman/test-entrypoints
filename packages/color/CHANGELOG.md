2.0.2 - New Changelog
