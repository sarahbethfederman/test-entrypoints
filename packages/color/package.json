{
  "name": "@lendi-ui/color",
  "version": "3.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^6.0.0",
    "styled-components": "^3"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^2.0.0"
  },
  "scripts": {
    "build:esm": "tsc --project tsconfig.esm.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:esm"
  }
}
