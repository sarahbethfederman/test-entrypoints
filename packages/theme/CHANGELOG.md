## [9.1.0] 2020-06-29 [FUNNEL-1473](https://creditandfinance.atlassian.net/browse/FUNNEL-1473)

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/breakpoint@6.0.0-beta.0
- Change `--lend-ui-size` to use new t-shirt breakpoints.

## [9.0.0] 2020-06-29

### Changed

- lendi-ui-size root size breakpoint changed to 576px instead of 577px

## [8.1.0] 2019-10-02 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- Enhancement LUI theme with default font-family for body and font-weight for heading.

## [8.0.1] 2019-9-29 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- added font family defaults based on theme

## [8.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

## [7.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

### Changed

- Color is now a type
- ColorPrefix is now a type
- theme.color object has been flattened
- Update typescript to version 3.5.2 and react to version 16.8.4

## [6.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [6.1.1] - 2019-05-10 - [HUB-137](https://creditandfinance.atlassian.net/browse/HUB-137)

### Added

- Add `-ms-clear` into theme injectGlobal to prevent cross icon in input by Edge and IE browser default style

## [6.1.0] - 2019-05-02 - [HUB-88](https://creditandfinance.atlassian.net/browse/HUB-88)

### Changed

- replace `border-radius: 6px` with select function in theme

## [6.0.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [6.0.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [6.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [5.0.0] - 2019-03-06 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Changed

- Added optional theme kind prop.

## [4.0.2] - 2019-02-25 - [BRAVO-511](https://creditandfinance.atlassian.net/browse/BRAVO-511)

### Changed

- Change warning colour to meet accessibility standards.

## [4.0.1]

### Added

- Added CHANGELOG file.
