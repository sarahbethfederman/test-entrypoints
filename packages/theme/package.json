{
  "name": "@lendi-ui/theme",
  "version": "9.1.0",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/breakpoint": "^5.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "scripts": {
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:cjs"
  }
}
