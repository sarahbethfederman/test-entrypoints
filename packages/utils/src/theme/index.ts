export type ColorKeys = '25' | '50' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900' | '950';

export type ColorsMap = { [key in ColorKeys]: string };

export type Typography = {
  heading: {
    fontFamily: string;
  };

  body: {
    fontFamily: string;
  };
};

export type Logo = {
  logoName: string;
};

export type Colors = {
  primary: ColorsMap;
  secondary: ColorsMap;
  tertiary: ColorsMap;
  shade: ColorsMap;
  info: ColorsMap;
  warn: ColorsMap;
  error: ColorsMap;
  success: ColorsMap;
};

export type ThemeMap = {
  typography: Typography;
  colors: Colors;
  logo: Logo;
};
