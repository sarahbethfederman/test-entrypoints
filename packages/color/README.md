# @lendi-ui/color

Color style with `bg`(background color), `fg`(font color) and `color`(every CSS property relate to color).

## Installation

```
yarn add @lendi-ui/color
```

## Usage

- `bg` for styling background-color.

```jsx
import { bg } from '@lendi-ui/color';
const Square = styled.div`
  width: 2em;
  height: 2em;
  ${bg('warn.500')};
`;
```

- `fg` for styling font-color.

```jsx
import { fg } from '@lendi-ui/color';
const WarningText = styled.span`
  ${fg('warn.500')};
`;
```

- `color` for styling every CSS porperty relate to color.

```jsx
import { color } from '@lendi-ui/color';
const Circle = styled.div`
  width: 0;
  border-radius: 1em;
  border: 1em solid ${color('primary.500')};
`;
```
