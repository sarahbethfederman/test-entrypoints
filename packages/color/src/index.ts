import { css } from 'styled-components';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type NameOrNameMap = BreakpointValue<string> | BreakpointValueMap<string>;

export function color(name: NameOrNameMap) {
  return select(`colors.${name}`);
}

export function fg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            color: ${color(n)};
          `
  );
}

export function bg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            background-color: ${color(n)};
          `
  );
}
