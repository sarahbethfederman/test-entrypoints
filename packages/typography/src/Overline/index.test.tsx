import * as React from 'react';
import { shallow } from 'enzyme';

import { theme } from '@lendi-ui/theme';
import { color as getColor } from '@lendi-ui/color';

import { Overline, OverlineSize, OverlineAlignment } from '..';

const sizes: OverlineSize[] = ['lg', 'md', 'sm'];
const alignments: OverlineAlignment[] = [undefined, 'left', 'center', 'right'];

describe('Overline', () => {
  sizes.forEach((size) => {
    it(`should render styles for size "${size}"`, () => {
      const wrapper = shallow(<Overline size={size} theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should render styles for the dark colorScheme', () => {
    const wrapper = shallow(<Overline size="lg" theme={theme} colorScheme="dark" />);
    expect(wrapper).toHaveStyleRule('color', `${getColor('shade.0')({ theme })}`);
  });

  it('should render styles for the light colorScheme', () => {
    const wrapper = shallow(<Overline size="lg" theme={theme} colorScheme="light" />);
    expect(wrapper).toHaveStyleRule('color', `${getColor('shade.400')({ theme })}`);
  });

  it('should render as a link with link styles when href prop is provided', () => {
    const wrapper = shallow(<Overline size="lg" theme={theme} href="#" />);
    expect(wrapper).toHaveStyleRule('color', `${getColor('primary.500')({ theme })}`);
    expect(wrapper.find('a')).toBeTruthy();
  });

  alignments.forEach((alignment) => {
    it(`should render styles for align "${alignment}"`, () => {
      const wrapper = shallow(<Overline size="lg" theme={theme} align={alignment} />);
      expect(wrapper).toHaveStyleRule('text-align', alignment);
    });
  });
});
