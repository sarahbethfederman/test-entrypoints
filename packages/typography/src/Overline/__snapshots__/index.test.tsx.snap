// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overline should render styles for size "lg" 1`] = `
<styled.span
  size="lg"
  theme={
    Object {
      "colors": Object {
        "error": Object {
          "100": "#ffd1da",
          "200": "#ffa4b6",
          "25": "#fff3f5",
          "300": "#ff7692",
          "400": "#ff496e",
          "50": "#ffe8ec",
          "500": "#ff1c4a",
          "600": "#cc163b",
          "700": "#99102c",
          "800": "#660b1d",
          "900": "#33050e",
          "950": "#190207",
        },
        "info": Object {
          "100": "#dae7f7",
          "200": "#b5d0ef",
          "25": "#f5f9fd",
          "300": "#91b9e8",
          "400": "#6ca2e0",
          "50": "#ecf3fb",
          "500": "#488bd9",
          "600": "#396fad",
          "700": "#2b5382",
          "800": "#1c3756",
          "900": "#0e1b2b",
          "950": "#070d15",
        },
        "primary": Object {
          "100": "#ccf2ed",
          "200": "#99e5db",
          "25": "#f2fbfa",
          "300": "#66d9c9",
          "400": "#32ccb7",
          "50": "#e5f8f6",
          "500": "#00c0a5",
          "600": "#009984",
          "700": "#007363",
          "800": "#004c42",
          "900": "#002621",
          "950": "#001310",
        },
        "secondary": Object {
          "100": "#cedee1",
          "200": "#9ebdc4",
          "25": "#f2f6f7",
          "300": "#6d9ca7",
          "400": "#3d7a8a",
          "50": "#e6eef0",
          "500": "#0d5a6d",
          "600": "#0a4857",
          "700": "#073641",
          "800": "#05242b",
          "900": "#021215",
          "950": "#01080a",
        },
        "shade": Object {
          "0": "#ffffff",
          "100": "#e3e3e3",
          "1000": "#000000",
          "200": "#c8c8c8",
          "25": "#f8f8f8",
          "300": "#adadad",
          "400": "#929292",
          "50": "#f1f1f1",
          "500": "#777777",
          "600": "#5f5f5f",
          "700": "#474747",
          "800": "#2f2f2f",
          "900": "#171717",
          "950": "#0b0b0b",
        },
        "success": Object {
          "100": "#dbf2cc",
          "200": "#b8e599",
          "25": "#f6fbf2",
          "300": "#94d966",
          "400": "#71cc32",
          "50": "#edf8e5",
          "500": "#4ec000",
          "600": "#3e9900",
          "700": "#2e7300",
          "800": "#1f4c00",
          "900": "#0f2600",
          "950": "#071300",
        },
        "tertiary": Object {
          "100": "#ffe2df",
          "200": "#ffc5c0",
          "25": "#fff7f7",
          "300": "#ffa9a1",
          "400": "#ff8c82",
          "50": "#fff0ef",
          "500": "#ff7063",
          "600": "#cc594f",
          "700": "#99433b",
          "800": "#662c27",
          "900": "#331613",
          "950": "#190b09",
        },
        "warn": Object {
          "100": "#fcddcc",
          "200": "#f9bc99",
          "25": "#fef6f2",
          "300": "#f69a66",
          "400": "#f37932",
          "50": "#fdeee5",
          "500": "#f05800",
          "600": "#c04600",
          "700": "#903400",
          "800": "#602300",
          "900": "#301100",
          "950": "#170800",
        },
      },
      "logo": Object {
        "logoName": "LendiLogo",
      },
      "typography": Object {
        "body": Object {
          "fontFamily": "'Open Sans', sans-serif",
        },
        "heading": Object {
          "fontFamily": "'Cabin', sans-serif",
        },
      },
    }
  }
/>
`;

exports[`Overline should render styles for size "md" 1`] = `
<styled.span
  size="md"
  theme={
    Object {
      "colors": Object {
        "error": Object {
          "100": "#ffd1da",
          "200": "#ffa4b6",
          "25": "#fff3f5",
          "300": "#ff7692",
          "400": "#ff496e",
          "50": "#ffe8ec",
          "500": "#ff1c4a",
          "600": "#cc163b",
          "700": "#99102c",
          "800": "#660b1d",
          "900": "#33050e",
          "950": "#190207",
        },
        "info": Object {
          "100": "#dae7f7",
          "200": "#b5d0ef",
          "25": "#f5f9fd",
          "300": "#91b9e8",
          "400": "#6ca2e0",
          "50": "#ecf3fb",
          "500": "#488bd9",
          "600": "#396fad",
          "700": "#2b5382",
          "800": "#1c3756",
          "900": "#0e1b2b",
          "950": "#070d15",
        },
        "primary": Object {
          "100": "#ccf2ed",
          "200": "#99e5db",
          "25": "#f2fbfa",
          "300": "#66d9c9",
          "400": "#32ccb7",
          "50": "#e5f8f6",
          "500": "#00c0a5",
          "600": "#009984",
          "700": "#007363",
          "800": "#004c42",
          "900": "#002621",
          "950": "#001310",
        },
        "secondary": Object {
          "100": "#cedee1",
          "200": "#9ebdc4",
          "25": "#f2f6f7",
          "300": "#6d9ca7",
          "400": "#3d7a8a",
          "50": "#e6eef0",
          "500": "#0d5a6d",
          "600": "#0a4857",
          "700": "#073641",
          "800": "#05242b",
          "900": "#021215",
          "950": "#01080a",
        },
        "shade": Object {
          "0": "#ffffff",
          "100": "#e3e3e3",
          "1000": "#000000",
          "200": "#c8c8c8",
          "25": "#f8f8f8",
          "300": "#adadad",
          "400": "#929292",
          "50": "#f1f1f1",
          "500": "#777777",
          "600": "#5f5f5f",
          "700": "#474747",
          "800": "#2f2f2f",
          "900": "#171717",
          "950": "#0b0b0b",
        },
        "success": Object {
          "100": "#dbf2cc",
          "200": "#b8e599",
          "25": "#f6fbf2",
          "300": "#94d966",
          "400": "#71cc32",
          "50": "#edf8e5",
          "500": "#4ec000",
          "600": "#3e9900",
          "700": "#2e7300",
          "800": "#1f4c00",
          "900": "#0f2600",
          "950": "#071300",
        },
        "tertiary": Object {
          "100": "#ffe2df",
          "200": "#ffc5c0",
          "25": "#fff7f7",
          "300": "#ffa9a1",
          "400": "#ff8c82",
          "50": "#fff0ef",
          "500": "#ff7063",
          "600": "#cc594f",
          "700": "#99433b",
          "800": "#662c27",
          "900": "#331613",
          "950": "#190b09",
        },
        "warn": Object {
          "100": "#fcddcc",
          "200": "#f9bc99",
          "25": "#fef6f2",
          "300": "#f69a66",
          "400": "#f37932",
          "50": "#fdeee5",
          "500": "#f05800",
          "600": "#c04600",
          "700": "#903400",
          "800": "#602300",
          "900": "#301100",
          "950": "#170800",
        },
      },
      "logo": Object {
        "logoName": "LendiLogo",
      },
      "typography": Object {
        "body": Object {
          "fontFamily": "'Open Sans', sans-serif",
        },
        "heading": Object {
          "fontFamily": "'Cabin', sans-serif",
        },
      },
    }
  }
/>
`;

exports[`Overline should render styles for size "sm" 1`] = `
<styled.span
  size="sm"
  theme={
    Object {
      "colors": Object {
        "error": Object {
          "100": "#ffd1da",
          "200": "#ffa4b6",
          "25": "#fff3f5",
          "300": "#ff7692",
          "400": "#ff496e",
          "50": "#ffe8ec",
          "500": "#ff1c4a",
          "600": "#cc163b",
          "700": "#99102c",
          "800": "#660b1d",
          "900": "#33050e",
          "950": "#190207",
        },
        "info": Object {
          "100": "#dae7f7",
          "200": "#b5d0ef",
          "25": "#f5f9fd",
          "300": "#91b9e8",
          "400": "#6ca2e0",
          "50": "#ecf3fb",
          "500": "#488bd9",
          "600": "#396fad",
          "700": "#2b5382",
          "800": "#1c3756",
          "900": "#0e1b2b",
          "950": "#070d15",
        },
        "primary": Object {
          "100": "#ccf2ed",
          "200": "#99e5db",
          "25": "#f2fbfa",
          "300": "#66d9c9",
          "400": "#32ccb7",
          "50": "#e5f8f6",
          "500": "#00c0a5",
          "600": "#009984",
          "700": "#007363",
          "800": "#004c42",
          "900": "#002621",
          "950": "#001310",
        },
        "secondary": Object {
          "100": "#cedee1",
          "200": "#9ebdc4",
          "25": "#f2f6f7",
          "300": "#6d9ca7",
          "400": "#3d7a8a",
          "50": "#e6eef0",
          "500": "#0d5a6d",
          "600": "#0a4857",
          "700": "#073641",
          "800": "#05242b",
          "900": "#021215",
          "950": "#01080a",
        },
        "shade": Object {
          "0": "#ffffff",
          "100": "#e3e3e3",
          "1000": "#000000",
          "200": "#c8c8c8",
          "25": "#f8f8f8",
          "300": "#adadad",
          "400": "#929292",
          "50": "#f1f1f1",
          "500": "#777777",
          "600": "#5f5f5f",
          "700": "#474747",
          "800": "#2f2f2f",
          "900": "#171717",
          "950": "#0b0b0b",
        },
        "success": Object {
          "100": "#dbf2cc",
          "200": "#b8e599",
          "25": "#f6fbf2",
          "300": "#94d966",
          "400": "#71cc32",
          "50": "#edf8e5",
          "500": "#4ec000",
          "600": "#3e9900",
          "700": "#2e7300",
          "800": "#1f4c00",
          "900": "#0f2600",
          "950": "#071300",
        },
        "tertiary": Object {
          "100": "#ffe2df",
          "200": "#ffc5c0",
          "25": "#fff7f7",
          "300": "#ffa9a1",
          "400": "#ff8c82",
          "50": "#fff0ef",
          "500": "#ff7063",
          "600": "#cc594f",
          "700": "#99433b",
          "800": "#662c27",
          "900": "#331613",
          "950": "#190b09",
        },
        "warn": Object {
          "100": "#fcddcc",
          "200": "#f9bc99",
          "25": "#fef6f2",
          "300": "#f69a66",
          "400": "#f37932",
          "50": "#fdeee5",
          "500": "#f05800",
          "600": "#c04600",
          "700": "#903400",
          "800": "#602300",
          "900": "#301100",
          "950": "#170800",
        },
      },
      "logo": Object {
        "logoName": "LendiLogo",
      },
      "typography": Object {
        "body": Object {
          "fontFamily": "'Open Sans', sans-serif",
        },
        "heading": Object {
          "fontFamily": "'Cabin', sans-serif",
        },
      },
    }
  }
/>
`;
